homeassistant/components/bryant_evolution/climate.py

```diff
@@ -0,0 +1,212 @@
+    call async_write_ha_state as well.
+    """
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self: ClimateEntity,
+        host: str,
+        system_id: int,
+        zone_id: int,
+        client: BryantEvolutionClient,
+    ) -> None:
+        """Initialize an entity from parts."""
+        self._client = client
+        self._attr_name = f"Bryant Evolution (System {system_id}, Zone {zone_id})"
+        self._attr_temperature_unit = UnitOfTemperature.FAHRENHEIT
+        self._attr_supported_features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE
+            | ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+            | ClimateEntityFeature.FAN_MODE
+            | ClimateEntityFeature.TURN_ON
+            | ClimateEntityFeature.TURN_OFF
+        )
+        self._enable_turn_on_off_backwards_compatibility = False
+        self._attr_unique_id = f"bryant_evolution_{host}_{system_id}_{zone_id}"
+
+    async def async_update(self) -> None:
+        """Update the entity state."""
+        self._attr_current_temperature = await self._client.read_current_temperature()
+        self._attr_fan_mode = await self._client.read_fan_mode()
+
+        self._attr_target_temperature = None
+        self._attr_target_temperature_high = None
+        self._attr_target_temperature_low = None
+        self._attr_hvac_mode = await self._read_hvac_mode()
+
+        # Set target_temperature or target_temperature_{high, low} based on mode.
+        match self._attr_hvac_mode:
+            case HVACMode.HEAT:
+                self._attr_target_temperature = (
+                    await self._client.read_heating_setpoint()
+                )
+            case HVACMode.COOL:
+                self._attr_target_temperature = (
+                    await self._client.read_cooling_setpoint()
+                )
+            case HVACMode.HEAT_COOL:
+                self._attr_target_temperature_high = (
+                    await self._client.read_cooling_setpoint()
+                )
+                self._attr_target_temperature_low = (
+                    await self._client.read_heating_setpoint()
+                )
+            case HVACMode.OFF:
+                pass
+            case _:
+                _LOGGER.error("Unknown HVAC mode %s", self._attr_hvac_mode)
+
+        # Note: depends on current temperature and target temperature low read
+        # above.
+        self._attr_hvac_action = await self._read_hvac_action()
+
+    async def _read_hvac_mode(self) -> HVACMode:
+        mode_and_active = await self._client.read_hvac_mode()
+        if not mode_and_active:
+            raise HomeAssistantError("Failed to read current HVAC mode")
+        mode = mode_and_active[0]
+        match mode.upper():
+            case "HEAT":
+                return HVACMode.HEAT
+            case "COOL":
+                return HVACMode.COOL
+            case "AUTO":
+                return HVACMode.HEAT_COOL
```

Comment:
why are you mapping "auto" to heat_cool instead of "auto"? (temperature range is supported by both)

Reply:
So, I read https://developers.home-assistant.io/docs/core/entity/climate/ and saw

HVACMode.HEAT_COOL	The device is set to heat/cool to a target temperature range.
HVACMode.AUTO	The device is set to a schedule, learned behavior, AI.

Also, looking at the const.py for the climate platform, AUTO says

"The temperature is set based on a schedule, learned behavior, AI or some other related mechanism. User is not able to adjust the temperature"

The Bryant definitely does not do AI or learned behavior. You can program a schedule, but it's still possible to adjust the temperature.

So, on balance it felt more like HEAT_COOL than AUTO.